### PR TITLE
Use JLayout for field file

### PR DESCRIPTION
**Pull request by dgrammatiko**
Separate Logic from output.
This is 100% B/C and allows templates to customise the core output of Joomla.
There shouldn't be any performance degradation since JLayouts are using caching.
It's a straight forward change, nothing magic here!

Response to the conversation: https://groups.google.com/forum/#!topic/joomla-dev-cms/I3onEgcwKZE

Notifying @chrisdavenport 

Testing
Try to upload a file in media manager or install a component/module/plugin by selecting a zip file

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on 569e7eeb39b2d91da17107f917133cafd0c223ca<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10063">issues.joomla.org/joomla-cms/10063</a>.</sub>

File: layouts/joomla/form/field/file.php [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @var   array    $inputType       Options available for this field.
+ * @var   array    $spellcheck      Options available for this field.
+ * @var   string   $accept          File types that are accepted.
+ */
+
+// Including fallback code for HTML5 non supported browsers.
+JHtml::_('jquery.framework');
+JHtml::_('script', 'system/html5fallback.js', false, true);
+?>
+Tste:
```
Comment (brianteeman) [inline layouts/joomla/form/field/file.php]: ????

File: layouts/joomla/form/field/file.php [hunk 2]
```diff
@@ -49,14 +49,15 @@
 JHtml::_('jquery.framework');
 JHtml::_('script', 'system/html5fallback.js', false, true);
 ?>
+Tste:
```
Comment (andrepereiradasilva) [inline layouts/joomla/form/field/file.php]: this `Tste:` is intended?

**Comment by andrepereiradasilva** [comment]
I have tested this item :red_circle: unsuccessfully on d53a5c9213e9535492eb0289c16ae70c9eedbf8b<br /><br />Besides the `Tste:` i tested this by adding to one of the joomla xmls.

```xml
<field
	name="testfile"
	type="file"
	label="custom label"
	description="custom description"
	size="40"
	accept="image/*"
	class="test-class1 test-class2"
	disabled="true"
	multiple="true"
	autofocus="true"
	required="true"
	onchange="alert('a');"
/>
```

Result:
```html
<input type="file"
	name="jform[testfile][]"
	id="jform_testfile"
	 required aria-required="true" />
```<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10063">issues.joomla.org/joomla-cms/10063</a>.</sub>

**Comment by dgt41** [comment]
Oops, my bad! Now everything should be fine:
screen shot 2016-04-23 at 19 21 25

**Comment by andrepereiradasilva** [comment]
@dgt41 thanks it works. for me is tested with success.
I would do it with clean HTML output (without tabs and line ends), but is working fine.

Before patch:
```html
<input type="file" name="jform[testfile][]" id="jform_testfile" accept="image/*" disabled class="test-class required" size="40" onchange="alert('a');" required aria-required="true" autofocus multiple />
```

After patch
```html
<input type="file"
	name="jform[testfile][]"
	id="jform_testfile"
	 size="40"	 accept="image/*"	 class="test-class required"	 multiple	 disabled	 autofocus	 onchange="alert('a');"	 required aria-required="true" />
```

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 4784cee91574682fe115ba246b688399d1180ead<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10063">issues.joomla.org/joomla-cms/10063</a>.</sub>

**Comment by chrisdavenport** [comment]
Looks good to me.  I like it.

**Comment by smz** [comment]
I have tested this item :white_check_mark: successfully on 4784cee91574682fe115ba246b688399d1180ead<br /><br />Way to go! :+1: 

But... in the JLayout, why not just:

~~~
echo '<input type="file" name="' . $name . '" id="' . $id . '"';
echo !empty($size) ? 'size="' . $size . '"' : '';
echo !empty($accept) ? ' accept="' . $accept . '"' : '';
echo !empty($class) ? ' class="' . $class . '"' : '';
echo !empty($multiple) ? ' multiple' : '';
echo $disabled ? ' disabled' : '';
echo $autofocus ? ' autofocus' : '';
echo !empty($onchange) ? ' onchange="' . $onchange . '"' : '';
echo $required ? ' required aria-required="true"' : '';
echo '/>';
~~~

that is without all the _opened-and-closed_ `<?php ... ?>`?
Using `echo` for the first part will also generate a tidier HTML without extra newlines and tabs...<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10063">issues.joomla.org/joomla-cms/10063</a>.</sub>

**Comment by dgt41** [comment]
@smz JLayouts are supposed to be HTML files so that front end devs can consume easier. I also try to concatenate those vars but I messed up the whole thing

**Comment by brianteeman** [comment]
Setting RTC and a milestone of 3.6

I assume you will now be working on the other fieldtypes<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10063">issues.joomla.org/joomla-cms/10063</a>.</sub>

**Comment by dgt41** [comment]
@brianteeman I will make PR's for all of them this week, but I need some time for calendar and other fields with JS in order to make sure that will be compatible with the new repeatable field. (I've got the code but haven't tested it against @fedik 's PR)

**Comment by brianteeman** [comment]
Great

**Pull request merged**